samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/device_composition.c

```diff
@@ -583,12 +666,48 @@ static void gen_onpowerup_set_unack(struct bt_mesh_model *model,
 	}
 }
 
+static s8_t onpowerup_set(struct bt_mesh_model *model,
+			  struct bt_mesh_msg_ctx *ctx,
+			  struct net_buf_simple *buf)
+{
+	u8_t onpowerup;
+	struct net_buf_simple *msg = model->pub->msg;
+	struct generic_onpowerup_state *state = model->user_data;
+
+	onpowerup = net_buf_simple_pull_u8(buf);
+
+	/* Here, Model specification is silent about tid implementation */
+
+	if (onpowerup > 0x02) {
+		return -1;
+	}
+	state->onpowerup = onpowerup;
+
+	/* Do some work here to save value of state->onpowerup on SoC flash */
+
+	if (model->pub->addr != BT_MESH_ADDR_UNASSIGNED) {
+		int err;
+
+		bt_mesh_model_msg_init(msg,
+				       BT_MESH_MODEL_OP_2(0x82, 0x12));
+		net_buf_simple_add_u8(msg, state->onpowerup);
+
+		err = bt_mesh_model_publish(model);
+		if (err) {
+			printk("bt_mesh_model_publish err %d\n", err);
+		}
+	}
+
+	return 0;
+}
+
 static void gen_onpowerup_set(struct bt_mesh_model *model,
 			      struct bt_mesh_msg_ctx *ctx,
 			      struct net_buf_simple *buf)
 {
-	gen_onpowerup_set_unack(model, ctx, buf);
-	gen_onpowerup_get(model, ctx, buf);
+	if (onpowerup_set(model, ctx, buf) == 0) {
+		gen_onpowerup_get(model, ctx, buf);
```

Comment:
The `onpowerup_set()` function will have changed the parsing state of `buf` so calling `gen_onpowerup_get()` here will cause reading past the end of the buffer. This is the second or third time I'm telling you this, please don't make me do it again :)

Your options are to either pre-parse the value and pass a `u8_t` instead of `net_buf_simple` to `onpowerup_set()`, or then (possibly better since you don't need to change `gen_onpowerup_get()`) use `net_buf_simple_save()` and `net_buf_simple_restore()` like I suggested earlier.

Reply:
working on it.

Reply:
I think what had done earlier is better than all this. 
Just to explain developers, why to do more complications ?

What going to happen if I add simply one extra tmp variable in struct for onoff state or onpowerup state ? It is by the way user defined data. 

I'm going to make things as it was. Kindly consider it.

Reply:
@jhedberg 

 I think what I had done earlier is better than all this. 
Just to explain developers, why to do more complications ?

What going to happen if I add simply one extra tmp variable in struct for onoff state or onpowerup state ? It is by the way user defined data. 

I'm going to make things as it was. Kindly consider it.

Reply:
@jhedberg 
I will make changes as per your suggestion in NEXT PR.  Kindly consider as of now.

Reply:
Done !!